docs/upload_specifications.ADR.md

```diff
@@ -0,0 +1,170 @@
+  file in the StS. The Narrative is expected to understand these types and map them to uploader
+  apps.
+2. The IDs of the app inputs from the `spec.json` file. 
+3. The corresponding human readable names of the app inputs from the `display.yaml` file.
+4. (and beyond) Import specifications. Each line corresponds to a single import.
+
+For Excel files, the first two rows may be hidden in any provided templates. Additionally,
+Excel files may contain multiple data types, one per tab.
+
+As part of this project we will deliver:
+1. CSV templates for each in scope app (e.g. the first 3 lines of the example file)
+2. An Excel template containing a tab for each in scope app
+3. A `README.md` file explaining how to use the templates.
+   * The `README.md` should include a link to rich upload specification documentation on the KBase
+     website once it is developed.
+   * Note: cover booleans which are not intuitive.
+
+These files will reside in the StS repo. As part of the front end effort, some means of
+delivering the templates and instructions to users will be developed.
+
+Currently, for all in scope apps, the inputs are all strings, numbers, or booleans. There are
+no unusual inputs such as grouped parameters or dynamic lookups. Including future upload apps
+with these features in CSV-based upload may require additional engineering.
+
+## User operations
+
+* The user uploads the upload specification files to the staging area along with all the files
+  inluded in the specification. 
+* The user selects the `Upload Specification` type for the specification files.
+  * The user may also select other files in the staging area to include in the upload along
+    with the files listed in the specification.
+    * The user *does not* have to select any files included in the specification.
+* The user clicks `Import Selected`.
+
+As such, a new type must be added to the StS: `upload_specification` with the title
+`Upload Specification`. *Nota bene*: It may be preferable to have the Narrative specify the
+titles to display in the staging area dropdown rather than the StS.
+
+## Narrative Bulk Upload cell operations
+
+When the bulk upload cell sees a `upload_specification` data type, it calls out to the StS
+endpoint detailed below to get the parsed specifications, and then autofills itself with
+those specifications.
+
+## Staging service upload specification endpoint
+
+The StS endpoint responds to an HTTP GET with a file list in a `files` URL parameter. It is
+extremely unlikely that there will be a large enough set of data types that URL length limits
+will be reached so this seems adequate. The StS will respond with the contents of the files
+parsed into a structure that is similar to that of the input spec, mapped by data type:
+
+```
+Headers:
+Authorization: <KBase token>
+
+GET /bulk_specification/?files=<file1.csv[,file2.csv,...]>
+
+Response:
```

Comment:
Does the user see this response? If so, my opinion on the comments on line 126 is to show the null entries. If not, I don't think it makes any UX difference.

Reply:
No, the narrative will accept the response and then use the data to initialize the upload cell.